sql/core/src/test/scala/org/apache/spark/sql/FilterPushdownBenchmark.scala

```diff
@@ -0,0 +1,243 @@
+    Native ORC Vectorized                         7080 / 7107          2.2         450.1       1.3X
+    Native ORC Vectorized (Pushdown)              6507 / 6552          2.4         413.7       1.4X
+
+    Select all rows (id IS NOT NULL):       Best/Avg Time(ms)    Rate(M/s)   Per Row(ns)   Relative
+    -----------------------------------------------------------------------------------------------
+    Parquet Vectorized                            9317 / 9366          1.7         592.4       1.0X
+    Parquet Vectorized (Pushdown)                 9316 / 9367          1.7         592.3       1.0X
+    Native ORC Vectorized                         7148 / 7210          2.2         454.5       1.3X
+    Native ORC Vectorized (Pushdown)              7092 / 7152          2.2         450.9       1.3X
+
+    Select all rows (id > -1):              Best/Avg Time(ms)    Rate(M/s)   Per Row(ns)   Relative
+    -----------------------------------------------------------------------------------------------
+    Parquet Vectorized                            9307 / 9353          1.7         591.7       1.0X
+    Parquet Vectorized (Pushdown)                 9303 / 9340          1.7         591.5       1.0X
+    Native ORC Vectorized                         7192 / 7249          2.2         457.2       1.3X
+    Native ORC Vectorized (Pushdown)              7182 / 7216          2.2         456.6       1.3X
+
+    Select all rows (id != -1):             Best/Avg Time(ms)    Rate(M/s)   Per Row(ns)   Relative
+    -----------------------------------------------------------------------------------------------
+    Parquet Vectorized                            9145 / 9328          1.7         581.4       1.0X
+    Parquet Vectorized (Pushdown)                 9320 / 9368          1.7         592.5       1.0X
+    Native ORC Vectorized                         7202 / 7230          2.2         457.9       1.3X
+    Native ORC Vectorized (Pushdown)              7170 / 7206          2.2         455.9       1.3X
+    */
+    benchmark.run()
+  }
+
+  def main(args: Array[String]): Unit = {
+    val numRows = 1024 * 1024 * 15
+    val width = 5
+    val mid = numRows / 2
+
+    withTempPath { dir =>
+      withTempTable("orcTable", "patquetTable") {
+        prepareTable(dir, numRows, width)
+
+        Seq("id IS NULL", s"$mid < id AND id < $mid").foreach { whereExpr =>
+          val title = s"Select 0 row ($whereExpr)".replace("id AND id", "id")
+          filterPushDownBenchmark(numRows, title, whereExpr)
+        }
+
+        Seq(
+          s"id = $mid",
+          s"id <=> $mid",
+          s"$mid <= id AND id <= $mid",
+          s"${mid - 1} < id AND id < ${mid + 1}"
+        ).foreach { whereExpr =>
+          val title = s"Select 1 row ($whereExpr)".replace("id AND id", "id")
+          filterPushDownBenchmark(numRows, title, whereExpr)
+        }
+
+        val selectExpr = (1 to width).map(i => s"LENGTH(c$i)").mkString("SUM(", "+", ")")
```

Comment:
Since the data is increased, I used `SUM(LENGTH(c1)+...)` instead of `*`.

Reply:
maybe simply `max(c1), max(c2), ...`?

Reply:
I see. I'll update and rerun.